source/common/upstream/thread_aware_lb_impl.h

```diff
@@ -27,6 +30,19 @@ class ThreadAwareLoadBalancerBase : public LoadBalancerBase, public ThreadAwareL
   public:
     virtual ~HashingLoadBalancer() = default;
     virtual HostConstSharedPtr chooseHost(uint64_t hash, uint32_t attempt) const PURE;
+    const std::string hashKey(HostConstSharedPtr host, bool use_hostname) {
```

Comment:
Sorry one last question. Is it possible for this to be a string_view vs. a string to avoid the copy? I think it might be possible but I'm not sure. Otherwise LGTM!

/wait-any